docs/ext/pydantic_autosummary/generate.py

```diff
@@ -0,0 +1,880 @@
+from sphinx.registry import SphinxComponentRegistry
+from sphinx.util import logging, rst
+from sphinx.util.inspect import getall, safe_getattr
+from sphinx.util.osutil import ensuredir
+from sphinx.util.template import SphinxTemplateLoader
+
+# ----------- pydantic_autosummary change
+from . import (
+    ImportExceptionGroup,
+    get_documenter,
+    import_by_name,
+    import_ivar_by_name,
+)
+from .pydantic import set_pydantic_model_fields
+
+# ----------/ pydantic_autosummary change
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence, Set
+    from gettext import NullTranslations
+
+    from sphinx.application import Sphinx
+    from sphinx.ext.autodoc import Documenter
+
+logger = logging.getLogger(__name__)
+
+
+class DummyApplication:
+    """Dummy Application class for sphinx-autogen command."""
+
+    def __init__(self, translator: NullTranslations) -> None:
+        self.config = Config()
+        self.registry = SphinxComponentRegistry()
+        self.messagelog: list[str] = []
+        self.srcdir = "/"
+        self.translator = translator
+        self.verbosity = 0
+        self._warncount = 0
+        self.warningiserror = False
+
+        self.config.add("autosummary_context", {}, "env", ())
+        self.config.add("autosummary_filename_map", {}, "env", ())
+        self.config.add("autosummary_ignore_module_all", True, "env", bool)
+
+    def emit_firstresult(self, *args: Any) -> None:
+        pass
+
+
+class AutosummaryEntry(NamedTuple):
+    name: str
+    path: str | None
+    template: str
+    recursive: bool
+
+
+def setup_documenters(app: Any) -> None:
+    from sphinx.ext.autodoc import (
+        AttributeDocumenter,
+        ClassDocumenter,
+        DataDocumenter,
+        DecoratorDocumenter,
+        ExceptionDocumenter,
+        FunctionDocumenter,
+        MethodDocumenter,
+        ModuleDocumenter,
+        PropertyDocumenter,
+    )
+
+    documenters: list[type[Documenter]] = [
+        ModuleDocumenter,
+        ClassDocumenter,
+        ExceptionDocumenter,
+        DataDocumenter,
+        FunctionDocumenter,
+        MethodDocumenter,
+        AttributeDocumenter,
+        DecoratorDocumenter,
+        PropertyDocumenter,
+    ]
+    for documenter in documenters:
+        app.registry.add_documenter(documenter.objtype, documenter)
+
+
+def _underline(title: str, line: str = "=") -> str:
+    if "\n" in title:
+        msg = "Can only underline single lines"
+        raise ValueError(msg)
+    return title + "\n" + line * len(title)
+
+
+class AutosummaryRenderer:
+    """A helper class for rendering."""
+
+    def __init__(self, app: Sphinx) -> None:
+        if isinstance(app, Builder):
+            msg = "Expected a Sphinx application object!"
+            raise ValueError(msg)
+
+        # ----------- pydantic_autosummary change
+        system_templates_path = [os.path.join(os.path.dirname(__file__), "templates")]
+        # ----------/ pydantic_autosummary change
```

Comment:
This changed the templates path to use the one embedded in the same directory over the one shipped with Sphinx.